access-token-management/src/AccessTokenManagement/AccessTokenHandlerOfT.cs

```diff
@@ -0,0 +1,104 @@
+/// <typeparam name="TToken">The type of access token that's added to the request. </typeparam>
+public sealed class AccessTokenHandler<TTokenRetriever, TToken> : DelegatingHandler
+    where TToken : ClientCredentialsToken
+    where TTokenRetriever : ITokenRetriever<TToken>
+{
+    private TTokenRetriever _tokenRetriever;
+    private readonly ISendRequestRetryPolicy _retryPolicy;
+    private readonly IDPopProofRequestHandler _dPoPProofRequestHandler;
+    private readonly ILogger<TTokenRetriever> _logger;
+
+    internal AccessTokenHandler(ISendRequestRetryPolicy retryPolicy,
+        TTokenRetriever tokenRetriever,
+        IDPopProofRequestHandler dPoPProofRequestHandler,
+        ILogger<TTokenRetriever> logger)
+    {
+        _retryPolicy = retryPolicy;
+        _dPoPProofRequestHandler = dPoPProofRequestHandler;
+        _logger = logger;
+        _tokenRetriever = tokenRetriever;
+    }
+
+    protected override HttpResponseMessage Send(HttpRequestMessage request, CancellationToken cancellationToken) =>
+        throw new NotSupportedException(
+            "The (synchronous) Send() method is not supported. Please use the async SendAsync variant. ");
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+        CancellationToken cancellationToken)
+    {
+        var response = await _retryPolicy.Handle(request, SendRequestWithToken, cancellationToken);
+
+        return response;
+    }
+
+    private async Task<HttpResponseMessage> SendRequestWithToken(
+        AccessTokenHandlerRequestData data,
+        CancellationToken cancellationToken)
+    {
+        var request = data.Request;
+        // Add a log scope that adds the Request URL to all subsequent log messages
+        using var logScope = _logger.BeginScope(
+            (OTelParameters.RequestUrl, request.RequestUri?.GetLeftPart(UriPartial.Path))
+        );
+
+        var token = await _tokenRetriever.GetToken(request, data.ForceTokenRefresh, cancellationToken);
+
+        var scheme = token.AccessTokenType ?? OidcConstants.AuthenticationSchemes.AuthorizationHeaderBearer;
+
+        if (!string.IsNullOrWhiteSpace(token.DPoPJsonWebKey))
+        {
+            // looks like this is a DPoP bound token, so try to generate the proof token
+            if (!await _dPoPProofRequestHandler.TryAcquireDPopProof(request, data.DPoPNonce, token, cancellationToken))
+            {
+                // failed or opted out for this request, to fall back to Bearer 
+                scheme = OidcConstants.AuthenticationSchemes.AuthorizationHeaderBearer;
+            }
+        }
+
+        // since AccessTokenType above in the token endpoint response (the token_type value) could be case insensitive, but
+        // when we send it as an Authorization header in the API request it must be case sensitive, we 
+        // are checking for that here and forcing it to the exact casing required.
```

Comment:
I don't think that we should be case sensitive. As per [RFC 9110: HTTP Semantics section 11.1](https://www.rfc-editor.org/rfc/rfc9110.html#name-authentication-scheme), authentication scheme names are case insensitive - and the latest draft of OAuth 2.1 has a funny set of examples of valid authorization headers: 
   *  Authorization: Bearer mF_9.B5f-4.1JqM
   *  Authorization: bearer mF_9.B5f-4.1JqM
   *  Authorization: BEARER mF_9.B5f-4.1JqM
   *  Authorization: bEaReR mF_9.B5f-4.1JqM

Reply:
This was logic that was built into the original code. If we can remove it, i'm all for it.